src/Authoring/WinRT.SourceGenerator/DiagnosticUtils.cs

```diff
@@ -124,7 +124,30 @@ private void CheckDeclarations()
                     CheckProperties(props, classId);
 
                     // check types -- todo: check for !valid types
-                    CheckMethods(publicMethods, classId);
+                    CheckMethods(publicMethods, classId);
+
+                    var iWinRTObject = model.Compilation.GetTypeByMetadataName("WinRT.IWinRTObject");

```

Comment:
would you consider moving this (the new logic) to a helper function and calling that here?

Reply:
sure. also made the thing about 'comment this code' automatic now for tests, should save us from wasting time when we forget to comment things out : )

Reply:
Great, thank you!